src/revwalk.c

```diff
@@ -664,6 +707,9 @@ void git_revwalk_sorting(git_revwalk *walk, unsigned int sort_mode)
 		walk->get_next = &revwalk_next_unsorted;
 		walk->enqueue = &revwalk_enqueue_unsorted;
 	}
+
+	if (sort_mode != GIT_SORT_NONE)
+		walk->limited = 1;
```

Comment:
This has the limitation that we will stay limited when somebody is switching sorting to `GIT_SORT_NONE` afterwards, again. But handling that case correctly would need another flag, so I guess that's okay

Reply:
Handling that would need something like `else if (!walk->did_hide)` which is not too bad, but would be quite an edge case where you're double-guessing what sorting you want for a single walk.